x/airdrop/types/airdrop.go

```diff
@@ -70,11 +79,21 @@ func (cr ClaimRecord) ValidateBasic() error {
 		i := 0
 		sum := uint64(0)
 		for ae, ca := range cr.ActionsCompleted {
+			// check enum bounds
+			kstr := fmt.Sprintf("ActionsCompleted[%d]", i)
 			if int(ae) >= len(Action_name) {
-				kstr := fmt.Sprintf("ActionsCompleted[%d]", i)
-				errors[kstr] = fmt.Errorf("enum out of bounds, expects [0-%d), got %d", len(Action_name), ae)
+				errors[kstr+" Enum"] = fmt.Errorf("%w, got %d", ErrActionOutOfBounds, ae)
 			}
+			// calc sum
 			sum += ca.ClaimAmount
+			// check completed time
+			if ca.CompleteTime.After(time.Now()) {
```

Comment:
this could be non-deterministic. time.Now() will differ between nodes - use ctx.BlockTime() instead.

Reply:
this is in ValidateBasic, which doesn't have access to ctx...
for the purposes here we only need to ensure that the CompleteTime is in the past.
Any recommendations?

Reply:
we could move this check from stateless to stateful validation...

Reply:
This is only for invariant checking? If so, we can probably let it slide as long as we comment in the code here the risk that _significant_ clock drift may cause nodes to fail invariant checking.